### PR TITLE
Add x11/wayland feature switches

### PR DESCRIPTION
**Pull request by DanySpin97**

### CODE REVIEW

**Comment by flightlessmango** [comment]
I think this was in before but removed because mangohud does not work on wayland for some reason

**Comment by telans** [comment]
segfaults for me

File: meson_options.txt [hunk 1]
```diff
@@ -1 +1,3 @@
 option('glibcxx_asserts', type : 'boolean', value : false)
+option('with_x11', type : 'feature', value : 'enabled')
```
Comment (jackun) [inline meson_options.txt]: Can't this be just a boolean?

**Comment by jackun** [comment]
Atleast hotkeys stuff needs to be \#ifdef'ed out if disabling x11

**Comment by jackun** [comment]
Merged manually.

**Pull request closed without merging**